jax_test3.py

```diff
@@ -55,65 +55,51 @@ def E_NN(N, Zs, *Rs):
     return e_nn
 
 
-def get_grad(coordinates, charges):
-    gradient = []
-    natoms = len(charges)
-    state = np.zeros(
-        2 + 3 * natoms, dtype=int
-    )  # makes an array that derv can work with
-    flat_coords = coordinates.reshape(3 * natoms)
-    for item in range(2, len(state)):
-        state[item] = 1  # change the array to the wanted derivation for the hessian
-        eval_E = derv(
-            E_NN,
-            [charges.size, charges, *flat_coords],
-            state,
-        )
-        gradient.append(eval_E)
-        state[item] = 0  # reset array
-
-    gradient = np.array(gradient)
-    gradient = gradient.reshape((natoms, 3))
-    return gradient
-
-
-# this will generate the full hessian
-def get_hessian(coordinates, charges):
-    natoms = len(charges)
-    state = np.zeros(2 + 3 * natoms, dtype=int)
-    hessian = np.zeros(
-        (natoms, 3) * 2
-    )  # reshaping the hessian gave errors so here's created a priori
-    # these will keep track whether we are on the x y or z coordinate
-    i_xyz = 0
-    a_xyz = 0
-    flat_coords = coordinates.reshape(3 * natoms)
-    for i in range(2, len(state)):
-        state[i] += 1  # change the item state
-        for a in range(2, len(state)):
-            state[a] += 1
+def derv_plus(current, stop, state, dervs, charges, flat_coords):
```

Comment:
```suggestion
def _derv_plus(current, stop, state, dervs, charges, flat_coords):
```
rationale: we can mark explicitly that this function is for internal use only by making its name start with an underscore.